adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -236,9 +236,12 @@ def stroke_extraction_from_matlab(self):
         - E is a k x 2 sequence of "graph edges"
         """
         eng = matlab.engine.start_matlab()
-        s = eng.genpath("./lightspeed")
+        file_dir = os.path.dirname(__file__)
+        s = eng.genpath(file_dir)
         eng.addpath(s, nargout=0)
-        s = eng.genpath("./BPL")
+        s = eng.genpath(os.path.join(file_dir, "lightspeed"))
+        eng.addpath(s, nargout=0)
+        s = eng.genpath(os.path.join(file_dir, "BPL"))
```

Comment:
Checked that it works mostly LGTM. One suggestion, inline the `s` variables:

```
        eng.addpath(eng.genpath(file_dir), nargout=0)
        eng.addpath(eng.genpath(os.path.join(file_dir, "lightspeed")), nargout=0)
        eng.addpath(eng.genpath(os.path.join(file_dir, "BPL")), nargout=0)
```

Reply:
Good idea, thanks. I also deleted the second `addpath()` calls since it looks like they're redundant ([genpath()](https://www.mathworks.com/help/matlab/ref/genpath.html) includes subdirectories).